pilot/pkg/proxy/envoy/v2/ep_filters.go

```diff
@@ -90,6 +92,8 @@ func EndpointsByNetworkFilter(push *model.PushContext, proxyNetwork string, endp
 						Value: weight,
 					},
 				}
+				// TODO: figure out a way to extract locality data from the gateway public endpoints in meshNetworks
+				gwEp.Metadata = util.BuildLbEndpointMetadata("", network, model.IstioMutualTLSModeLabel, push)
```

Comment:
Why hard coded this , i think it means always initiate mtls to this gateway endpoint?

Reply:
if the cluster's transport socket match isn't set. this will take no effect?

Reply:
Yep. If the transport socket match is not set, this wont take effect. And we wont set the transport socket match if user creates a destination rule with tls disable. That said, there is no way to do split horizon over plain text. You need mTLS so that you can distinguish the target envoy cluster at the Gateway (i.e. auto_passthrough tls mode at the gateway).